flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/expressions/catalog/FunctionDefinitionCatalog.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.expressions.catalog;
+
+import org.apache.flink.table.expressions.FunctionDefinition;
+
+/**
+ * Catalog of functions that can resolve function's name to {@link FunctionDefinition}.
+ */
+public interface FunctionDefinitionCatalog {
+
+	/**
+	 * Lookup a function by name and return the {@link FunctionDefinition}.
+	 */
+	FunctionDefinition lookupFunction(String name);
```

Comment:
When defining the interface, can you also document the current resolution behavior? I think we are case insensitive right now?